service/common/src/main/java/org/apache/polaris/service/persistence/InMemoryPolarisMetaStoreManagerFactory.java

```diff
@@ -59,8 +58,9 @@ public InMemoryPolarisMetaStoreManagerFactory(
     this.storageIntegration = storageIntegration;
   }
 
-  public void onStartup(RealmContextConfiguration realmContextConfiguration) {
-    bootstrapRealmsAndPrintCredentials(realmContextConfiguration.realms());
+  @Override
+  public void initializeForService(@Nonnull List<String> realmIds, @Nonnull String defaultRealmId) {
```

Comment:
It looks like `defaultRealmId` is not used. Why have it as a parameter?

Also, why not call `bootstrapRealms` from `QuarkusProducers`? In the latter case, it should be possible to inject root credentials from Quarkus config (short term) or secret providers (long term). WDYT?

Reply:
From my POV, if we have to add a public API method just to get rid of a type check in "producers", I'd rather not change interfaces, but make produces call the existing `bootstrapRealms` method.

We could have a config like `polaris.persistence.auto-bootstrap.type` with default `in-memory` instead of type checks... WDYT?

Reply:
Maybe in a follow-up?

Reply:
I'm not sure I understand :thinking: if we go with calling `bootstrapRealms` from `QuarkusProducers` then the `initializeForService` method added in this PR is kind of not needed, WDYT?